csrc/scheduler/vectorize_helper.cpp

```diff
@@ -818,46 +818,51 @@ size_t getVectorizationFactor(
     return 1;
   }
 
-  size_t max_vec_size = SchedulerRuntimeInfo::max_alignment_size_in_byte;
+  int64_t max_vec_size = SchedulerRuntimeInfo::max_alignment_size_in_byte;
+  const auto& tv_to_inner_size_map = vectorize_maps_entry.get().at(break_point);
 
   for (auto inp_or_out : vectorizable_inputs_outputs) {
-    auto dtype_size =
+    // factor <= max_factor / dtype_size
+    const auto dtype_size =
         dataTypeSize(inp_or_out->dtype(), runtime_info.getIndexType());
     max_vec_size = std::min(
         max_vec_size,
         SchedulerRuntimeInfo::max_alignment_size_in_byte / dtype_size);
-    auto alignment_size = runtime_info.getAlignmentSize(inp_or_out);
+
+    // factor <= alignment / dtype_size
+    int64_t alignment_size = (int64_t)runtime_info.getAlignmentSize(inp_or_out);
     TORCH_INTERNAL_ASSERT(alignment_size % dtype_size == 0);
-    auto new_max_vec_size = std::min(max_vec_size, alignment_size / dtype_size);
-    max_vec_size = new_max_vec_size;
-  }
+    max_vec_size = std::min(max_vec_size, alignment_size / dtype_size);
 
-  auto tv_to_inner_size_map = vectorize_maps_entry.get().at(break_point);
-  // Initialize to max the tensors could support.
-  size_t max_supported_vector_size = max_vec_size;
-  for (auto inp_or_out : vectorizable_inputs_outputs) {
+    // factor <= projected_extent
     auto inner_size_it = tv_to_inner_size_map.find(inp_or_out);
-    auto inner_size_val = inner_size_it != tv_to_inner_size_map.end()
-        ? inner_size_it->second
-        : inp_or_out->container()->oneVal();
+    if (inner_size_it == tv_to_inner_size_map.end()) {
+      // If we don't have info for a tensor that is supposed to be
+      // vectorized, that means the tensor has no projected
+      // vectorizable extent, i.e., not vectorizable.
+      // TODO: Instead of competely disabling vectorization for all
+      // tensors, just disable the problematic tensor and keep the
+      // other tensors vectorized
+      return 1;
+    }
```

Comment:
Is the condition possible to be true? I thought it should never be true, if yes, then it is a bug of `vectorizable_inputs_outputs`.

Reply:
I thought the same, so I initially inserted an assertion there and got these two failures:

```
[  FAILED  ] NVFuserTest.FusionSegmentReduceSoftmax_CUDA
[  FAILED  ] NVFuserTest.FusionReshapeMagicSchedule9_CUDA
```

Here's the fusion of `FusionSegmentReduceSoftmax`:

```
Inputs:
  T2_g[ iS6{i0}, iS7{i2}, rS8{i3} ], double
Outputs:
  T10_g[ iS23{i0}, iS24{i2} ], double

%kernel_math {
T3_l[ iS9{i0}, rS10{i2} ]
   = reduction( T2_g[ iS6{i0}, iS7{i2}, rS8{i3} ], op = fmax, initial value = double(-inf), allreduce = false )
T4_l[ iS11{i0}, bS12{1} ]
   = broadcast( T3_l[ iS9{i0}, rS10{i2} ] )
T5_l[ iS13{i0}, iS14{i2} ]
   = T2_g[ iS6{i0}, iS7{i2}, rS8{i3} ]
   - T4_l[ iS11{i0}, bS12{1} ];
T6_g[ iS15{i0}, iS16{i2} ]
   = exp(T5_l[ iS13{i0}, iS14{i2} ]);
T7_g[ iS17{i0}, rS18{i2} ]
   = reduction( T6_g[ iS15{i0}, iS16{i2} ], op = add, initial value = double(0), allreduce = false )
T8_l[ iS19{i0}, bS20{1} ]
   = broadcast( T7_g[ iS17{i0}, rS18{i2} ] )
T9_g[ iS21{i0}, bS22{1} ]
   = reciprocal(T8_l[ iS19{i0}, bS20{1} ]);
T10_g[ iS23{i0}, iS24{i2} ]
   = T6_g[ iS15{i0}, iS16{i2} ]
   * T9_g[ iS21{i0}, bS22{1} ];
}
```

The reference tensor is `T2`, the break point is 2, so the only reference root domain is `rS8{i3}`. Since it's a reduction domain, nothing is propagated to the output, and that's why there's no mapping for `T10`.

I didn't think about this further, but I'm now wondering why the break point is 2 instead of 0 or 1.

Reply:
> I didn't think about this further, but I'm now wondering why the break point is 2 instead of 0 or 1.

Yeah, strange. But anyway, let's merge the PR as is now, and leave the fix to a separate PR.